chord_metadata_service/chord/ingest.py

```diff
@@ -412,54 +427,154 @@ def _get_output_or_raise(workflow_outputs, key):
     return workflow_outputs[key]
 
 
+DRS_URI_SCHEME = "drs"
+FILE_URI_SCHEME = "file"
+HTTP_URI_SCHEME = "http"
+HTTPS_URI_SCHEME = "https"
+
+
+def _workflow_http_download(tmp_dir: str, http_uri: str) -> str:
+    # TODO: Sanity check: no external insecure HTTP calls
+    # TODO: Disable HTTPS cert check in debug mode
+    # TODO: Handle response exceptions
+
+    r = requests.get(http_uri)
+
+    if not r.ok:
+        raise IngestError(f"HTTP error encountered while downloading ingestion URI: {http_uri}")
+
+    data_path = f"{tmp_dir}ingest_download_data"
+
+    with open(data_path, "wb") as df:
+        df.write(r.content)
+
+    return data_path
+
+
+@contextlib.contextmanager
+def _workflow_file_output_to_path(file_uri: str):
+    # TODO: Should be able to download from DRS instead of using file URIs directly
+
+    parsed_file_uri = urlparse(file_uri)
+
+    if parsed_file_uri.scheme == FILE_URI_SCHEME:  # File URI
+        yield parsed_file_uri.path
+        return
+
+    if parsed_file_uri.scheme == "":  # File path with no URI scheme
+        yield parsed_file_uri.path
+        return
+
+    # From here on out, we're dealing with downloads - check to make sure we
+    # have somewhere to put the temporary files.
+
+    should_del = False
+    tmp_dir = settings.SERVICE_TEMP
+
+    if tmp_dir is None:
+        tmp_dir = tempfile.mkdtemp()
+        should_del = True
+
+    if not os.access(tmp_dir, os.W_OK):
+        raise IngestError(f"Directory does not exist or is not writable: {tmp_dir}")
+
+    try:
+        tmp_dir = tmp_dir.rstrip("/") + "/"
+
+        if parsed_file_uri.scheme == DRS_URI_SCHEME:  # DRS object URI
+            drs_obj = fetch_drs_record_by_uri(file_uri, settings.DRS_URL)
+
+            file_access = get_access_method_of_type(drs_obj, "file")
+            if file_access:
+                yield urlparse(file_access["access_url"]["url"]).path
+                return
+
+            # TODO: Some mechanism to do this with auth
+            http_access = get_access_method_of_type(drs_obj, "http")
+            if http_access:
+                # TODO: Handle DRS headers field if available - how to do this with grace and compatibility with
+                #  Bento's auth system?
+                yield _workflow_http_download(tmp_dir, http_access["access_url"]["url"])
+                return
+
+            # If we get here, we have a DRS object we cannot handle; raise an error.
+            raise IngestError(f"Cannot handle DRS object {file_uri}: No file or http access methods")
+
+        elif parsed_file_uri.scheme in (HTTP_URI_SCHEME, HTTPS_URI_SCHEME):
+            yield _workflow_http_download(tmp_dir, file_uri)
+
+        else:
+            # If we get here, we have a scheme we cannot handle; raise an error.
+            raise IngestError(f"Cannot handle workflow output URI scheme: {parsed_file_uri.scheme}")
```

Comment:
The tests in chord/tests/test_ingest.py fail with this error now because the file path is `example_phenopacket.json` and it has no `parsed_file_uri.scheme`. What do you think about adding another condition here that can handle paths that are just filenames?

Reply:
That's a good idea, I'll do that - for some reason I thought this was handled but I guess I didn't do it right, thanks for pointing it out!